### PR TITLE
`near-chain`: Fix `clippy::redundant-clone` issues.

### PR DESCRIPTION
**Pull request by pmnoxx**
Fix 'clippy::redundant-clone' clippy warnings in `near-chain`.

See https://github.com/near/nearcore/pull/5571

### CODE REVIEW

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -110,9 +110,7 @@ pub trait ChainStoreAccess {
     ) -> Result<ShardChunk, Error> {
         let shard_chunk_result = self.get_chunk(&header.chunk_hash());
         match shard_chunk_result {
-            Err(_) => {
-                return Err(ErrorKind::ChunksMissing(vec![header.clone()]).into());
-            }
+            Err(_) => Err(ErrorKind::ChunksMissing(vec![header.clone()]).into()),
```
Comment (bowenwang1996) [inline chain/chain/src/store.rs:113]: wait I don't think the transformation here is semantic preserving

Comment (pmnoxx) [reply chain/chain/src/store.rs:113]: The result of `match shard_chunk_result {` is returned to `match shard_chunk_result {`
then to ` fn get_chunk_clone_from_header(`

Comment (pmnoxx) [reply chain/chain/src/store.rs:113]: The result is preserved, by we can get @matklad opinion on this is you prefer.

Comment (bowenwang1996) [reply chain/chain/src/store.rs:113]: > The result of match shard_chunk_result { is returned to match shard_chunk_result {
then to fn get_chunk_clone_from_header(

I don't think that is the case. See official rust doc here https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html

Comment (pmnoxx) [reply chain/chain/src/store.rs:113]: This is a simpler case than the one in the doc. 
@matklad Which code style should we use from the two above:
A) This is what we have in the code
```
fn multiply(first_number_str: &str, second_number_str: &str) -> Result<i32, ParseIntError> {
   match first_number_str.parse::<i32>() {
        Ok(first_number)  => first_number,
        Err(e) => return Err(e),
    };
}
```
B) This is how we can write it
```
fn multiply(first_number_str: &str, second_number_str: &str) -> Result<i32, ParseIntError> {
   match first_number_str.parse::<i32>() {
        Ok(first_number)  => first_number,
        Err(e) => e,
    };
}
```

@bowenwang1996 No that, there is `let match = `.
If you think that this change is not correct, we should open an issue with `clippy`. https://github.com/rust-lang/rust-clippy

Comment (pmnoxx) [reply chain/chain/src/store.rs:113]: @bowenwang1996 Here is the warning from `clippy`:
```
warning: unneeded `return` statement
   --> chain/chain/src/store.rs:114:17
    |
114 |                 return Err(ErrorKind::ChunksMissing(vec![header.clone()]).into());
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove `return`: `Err(ErrorKind::ChunksMissing(vec![header.clone()]).into())`
    |
    = note: `-W clippy::needless-return` implied by `-W clippy::all`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
```

Comment (pmnoxx) [reply chain/chain/src/store.rs]: I removed this change from the PR, as it doesn't fix the clone() issue.

Comment (pmnoxx) [reply chain/chain/src/store.rs]: This belongs to a different PR.

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -2278,53 +2278,49 @@ impl Chain {
...
-            Ok(hash.clone())
+            Ok(*hash)
+        } else if level == 0 {
+            let maybe_hash = if index >= tree_size {
+                None
+            } else {
+                Some(*self.mut_store().get_block_hash_from_ordinal(index)?)
+            };
+            tree_nodes.insert((index, level), maybe_hash);
+            Ok(maybe_hash)
```
Comment (matklad) [inline chain/chain/src/chain.rs:2289]: @pmnoxx could you carefully check that this doesn't do anything except removal of the clones?

**Review by yordanmadzhunkov — APPROVED** [review]

File: chain/chain/src/chain.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/chain/src/chain.rs]: @matklad It does now xD.

Comment (pmnoxx) [reply chain/chain/src/chain.rs]: @matklad Thanks, for noticing. I rewrote the change, to only fix `clippy::redundant-clone` issues.

**Pull request merged**